### PR TITLE
Add coin-or stack

### LINKED ISSUE #1991: Add coin-or stack
The whole coin-or stack consists of compiled packages that also have run-exports.

As recommended by @wolfv here (https://github.com/conda-forge/staged-recipes/pull/16147#discussion_r716940253) this issue is to discuss whether it makes sense to add these packages.

Coin-or stack:

* https://github.com/conda-forge/coin-or-clp-feedstock
* https://github.com/conda-forge/coin-or-utils-feedstock
* https://github.com/conda-forge/coin-or-cgl-feedstock
* https://github.com/conda-forge/coin-or-cbc-feedstock
* https://github.com/conda-forge/coin-or-osi-feedstock

The author is @tkralphs.

This will also be important with the upcoming release of coin-or-clp 1.17.7 for which the stack will have to be rebuilt.

### PR DESCRIPTION
**Pull request by BastianZim**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

Closes #1991

cc @wolfv

### CODE REVIEW

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -365,6 +365,16 @@ cairo:
   - 1.16
 cfitsio:
   - 3.470
+coin-or-cbc:
+  - 2.10.5
+coin-or-cgl:
+  - 0.60.3
+coin-or-clp:
+  - 1.17.4
```
Comment (BastianZim) [inline recipe/conda_build_config.yaml:373]: The current version is 1.17.6 but that has some bugs and at least `or-tools` has a pin to 1.17.4, not sure how to handle this.

File: recipe/conda_build_config.yaml [hunk 2]
```diff
@@ -365,6 +365,16 @@ cairo:
   - 1.16
 cfitsio:
   - 3.470
+coin-or-cbc:
```
Comment (isuruf) [inline recipe/conda_build_config.yaml:368]:
```suggestion
coin_or_cbc:
```

Comment (BastianZim) [reply recipe/conda_build_config.yaml:368]: For everyone?

Comment (isuruf) [reply recipe/conda_build_config.yaml:368]: yes

**Comment by isuruf** [comment]
We should first make `coincbc` package a meta package before using the new `coin-or-cbc` package.

**Comment by BastianZim** [comment]
@isuruf Should `coincbc` be an output of `coin-or-cbc` and `coincbc` be archived, or should `coincbc` be empty and depend on `coin-or-cbc`?

**Comment by isuruf** [comment]
First option is preferred.

**Comment by BastianZim** [comment]
Has been introduced in https://github.com/conda-forge/coin-or-cbc-feedstock/pull/11

**Comment by BastianZim** [comment]
@isuruf Would you mind merging this now?

File: recipe/conda_build_config.yaml [hunk 3]
```diff
@@ -365,6 +365,16 @@ cairo:
   - 1.16
 cfitsio:
   - 3.470
+coin_or_cbc:
+  - 2.10
```
Comment (isuruf) [inline recipe/conda_build_config.yaml:369]:
```suggestion
coin_or_cbc:
  - 2.10
coincbc:
  - 2.10
```

**Comment by BastianZim** [comment]
Thank you!

**Pull request merged**